standard/interfaces.md

```diff
@@ -216,22 +218,33 @@ The members of an interface are the members inherited from the base interfaces a
+An interface declaration declares zero or more members. The members of an interface shall be constants, fields, methods, properties, events, indexers, operators, constructors, and types, some of which may be instance, others static, as described in the subclauses for each interface member kind.
+
+All interface members implicitly have public access; however, an explicit access modifier ([§7.5.2](basic-concepts.md#752-declared-accessibility)) is permitted.
+
+An interface function member whose declaration includes a body is an implicitly `virtual` member unless the `sealed` or `private` modifier is used. The `virtual` modifier may be used on a function member that would otherwise be implicitly `virtual`. Similarly, although `abstract` is implied for interface function members without bodies, that modifier may be given explicitly. A non-virtual function member may be declared using the `sealed` keyword.
 
-All interface members implicitly have public access. It is a compile-time error for interface member declarations to include any modifiers.
+A `private` or `sealed` function member of an interface shall have a body. A `private` function member shall not have the modifier `sealed`.
+
+An explicitly implemented function member shall not have the modifier `sealed`.
 
 An *interface_declaration* creates a new declaration space ([§7.3](basic-concepts.md#73-declarations)), and the type parameters and *interface_member_declaration*s immediately contained by the *interface_declaration* introduce new members into this declaration space. The following rules apply to *interface_member_declaration*s:
 
 - The name of a type parameter in the *variant_type_parameter_list* of an interface declaration shall differ from the names of all other type parameters in the same *variant_type_parameter_list* and shall differ from the names of all members of the interface.
-- The name of a method shall differ from the names of all properties and events declared in the same interface. In addition, the signature ([§7.6](basic-concepts.md#76-signatures-and-overloading)) of a method shall differ from the signatures of all other methods declared in the same interface, and two methods declared in the same interface shall not have signatures that differ solely by `in`, `out`, and `ref`.
-- The name of a property or event shall differ from the names of all other members declared in the same interface.
+- The name of a method shall differ from the names of all other kinds of function members declared in the same interface. In addition, the signature ([§7.6](basic-concepts.md#76-signatures-and-overloading)) of a method shall differ from the signatures of all other methods declared in the same interface, and two methods declared in the same interface may not have signatures that differ solely by `in`, `out`, and `ref`.
```

Comment:
Does "function members" include classes? For instance, this does not compile:
```cs
interface I
{
    void Foo();
    class Foo { }
}
```

Reply:
If "function members" _does_ include classes then this line above is incorrect because a nested class may have both the `private` and `sealed` modifiers:

> A `private` or `sealed` function member of an interface shall have a body. A `private` function member shall not have the modifier `sealed`.